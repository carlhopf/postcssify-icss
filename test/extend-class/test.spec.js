const browserify = require('browserify');
const rawBody = require('raw-body');
const postcssify = require('../../index');
const path = require('path');
const util = require('util');
const vm = require('vm');
const assert = require('assert');

describe('extend-class', function () {
	var jsStream;
	var jsStr;
	var cssStream;
	var onFiles = [];
	
	beforeEach(() => {
		var bundle = browserify({
			// assign toplevel module.exports to window['standalone']
			standalone: 'standalone',
		});
		
		bundle.plugin(postcssify, {
			before: [
				require('postcss-nested'),
			],
		});
		
		bundle.add(path.join(__dirname, 'index.js'));
		bundle.on('css stream', (stream) => { cssStream = stream; });
		bundle.on('file', (file) => { onFiles.push(file); });
		
		jsStream = bundle.bundle();
	});
	
	beforeEach((cb) => {
		rawBody(jsStream, (err, buf) => {
			expect(err).toBeFalsy();
			expect(buf.toString('utf8')).toMatchSnapshot();
			jsStr = buf.toString('utf8');
			cb(null);
		});
	});
	
	beforeEach((cb) => {
		rawBody(cssStream, (err, buf) => {
			expect(err).toBeFalsy();
			expect(buf.toString('utf8')).toMatchSnapshot();
			cb(null);
		});
	});
	
	test('defined', () => {
		const sandbox = {};
		const script = new vm.Script(jsStr);
		const context = new vm.createContext(sandbox);
		script.runInContext(context);
		
		assert(context);
		assert(context.standalone);
		
		expect(context.standalone.button).toBeTruthy();
		expect(context.standalone.extendButton).toBeTruthy();
		
		expect(context.standalone.extendButton)
			.not.toBe(context.standalone.button);
	});
});
