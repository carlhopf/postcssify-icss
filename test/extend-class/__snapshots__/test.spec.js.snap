// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extend-class defined 1`] = `
"(function(f){if(typeof exports===\\"object\\"&&typeof module!==\\"undefined\\"){module.exports=f()}else if(typeof define===\\"function\\"&&define.amd){define([],f)}else{var g;if(typeof window!==\\"undefined\\"){g=window}else if(typeof global!==\\"undefined\\"){g=global}else if(typeof self!==\\"undefined\\"){g=self}else{g=this}g.standalone = f()}})(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require==\\"function\\"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error(\\"Cannot find module '\\"+o+\\"'\\");throw f.code=\\"MODULE_NOT_FOUND\\",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require==\\"function\\"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
module.exports = {\\"extendButton\\":\\"button__button---1LIs-\\",\\"button\\":\\"index__button---R0coX\\"}
},{}],2:[function(require,module,exports){
module.exports = require('./index.css');

},{\\"./index.css\\":1}]},{},[2])(2)
});"
`;

exports[`extend-class defined 2`] = `
".button__button---1LIs- {
  background-color: red;
}
/* overrides background-color from button.css */
.button__button---1LIs- {
  background-color: green;
}

/* new local class, not related to button.css */
.index__button---R0coX {
  font-size: 1em;
}
"
`;
